src/item.cpp

```diff
@@ -5778,22 +5778,68 @@ bool item::process_food( player * /*carrier*/, const tripoint &pos )
-    diff_cold = diff_cold > 10 ? 10 : diff_cold;
-    // environment temperature applies COLD/FROZEN flags to food
-    if( item_local_temp <= FRIDGE_TEMPERATURE ) {
-        g->m.apply_in_fridge( *this, item_local_temp );
-    } else if ( item_tags.count( "FROZEN" ) > 0 && item_counter > diff_freeze ) {
-        item_counter -= diff_freeze;
-    } else if( item_tags.count( "COLD" ) > 0 && item_counter > diff_cold ) {
-        item_counter -= diff_cold;
+    const auto temp = g->get_temperature( pos );
+    unsigned int diff_freeze = abs( temp - FREEZING_TEMPERATURE ) / 5 ;
+    diff_freeze = clamp( diff_freeze, static_cast<unsigned int>(0) , static_cast<unsigned int>(3) ); //effective 1-4
+
+    // div by 5 means every 5 degrees of difference equals 1 point of ratio
+    unsigned int diff_cold = abs( temp - FRIDGE_TEMPERATURE ) / 5;
+    diff_cold = clamp( diff_cold, static_cast<unsigned int>(0), static_cast<unsigned int>(3) ); //effective 1-4
+
+    // exclusions for items being actively cooled to be affected by outside temperatures
+    //const optional_vpart_position vp = g->m.veh_at( pos );
+    const optional_vpart_position vp = g->m.veh_at( pos );
+    bool in_active_cooler = false;
+    bool fridge_here = false;
+    bool freezer_here = false;
+    if( vp ) {
+        fridge_here = vp->vehicle().has_part( pos, "FRIDGE", true );
+        freezer_here = vp->vehicle().has_part( pos, "FREEZER", true );
+        if(fridge_here) { add_msg( "Fridge" ); } else {add_msg( "NOT Fridge" );}
+        if(freezer_here) { add_msg( "Freezer" ); } else {add_msg( "NOT Freezer" );}
+    }
+
+    if( vp && ( fridge_here || freezer_here ) ) {
+        std::vector<vehicle_part *> parts = vp->vehicle().get_parts( pos, "FRIDGE", true );
+        if( parts.empty() ) { 
+            parts = vp->vehicle().get_parts( pos, "FREEZER", true );
+        }
+        if( parts.empty() ) { add_msg( "PARTS EMPTY" ); }
+        vehicle_part *part = parts.front();
+        vehicle_stack vit = vp->vehicle().get_items( vp->vehicle().index_of_part( part ) );
+        if( vit.empty() ) { add_msg( "ITEMS EMPTY" ); }
+        for( size_t iter = 0; iter < vit.size(); iter++ ) {
+            item *itm = &vit[iter];
+            if( itm == this ) {
+                in_active_cooler = true;
+                add_msg( "PING" );	
+            } else { add_msg( "PONG %s", itm->tname() );	}
+        }
+    }
+    
+    // environment temperature applies COLD/FROZEN flags to food unless in fridge/freezer
+    if( !in_active_cooler ) {
+        if( temp <= FRIDGE_TEMPERATURE ) {
+            add_msg( "NOT_IN_COOLER" );	
+            g->m.apply_in_fridge( *this, temp );
+            add_msg( "Temp: %s", temp );	
+        } else if ( item_tags.count( "FROZEN" ) > 0 && item_counter > diff_freeze ) {
+            item_counter -= diff_freeze; // thaw
+        } else if( item_tags.count( "COLD" ) > 0 && item_counter > diff_cold ) {
+            item_counter -= diff_cold; // get warm
+        }
```

Comment:
From what I can tell all of this item_counter stuff is already handled in apply_in_fridge, why do we need to do it all again here?

If that's correct then I think you can replace these if/else blocks with one call to g->m.apply_in_fridge and using an appropriate var for `temp`

Reply:
Answer is simple: apply_in_fridge() is responsible for cooling down items, while this is the scope of process_food() where counters are removed so it is a place where opposite process of warming up takes place.

Reply:
Gotcha, thanks for the other fixups and whatnot.  I still think L@5845-5862 can be made a little nicer somehow, but it can be lived with.  Labels on the debug code are nice, thanks.

As an idea then, you could have `apply_in_fridge` take care of the heating process too, and in that case I would rename it to something like `apply_temp_diff`

Reply:
I'll certainly check it out, as this might be a potential walk-around the problems I had with vehicle items.